### ISSUE TITLE
Scripting: TypeInitialization Exception upon any usage of Scripting Objects

### ISSUE THREAD

**Issue by RonPenton** [dotnet/roslyn#2856]
I'm running an ASP.NET 5 application, and I've included the latest nightly build from the MyGet NuGet package:  

    "Microsoft.CodeAnalysis.Scripting": "1.0.0-rc3-20150514-02",

The code I am trying to execute:

    ScriptState state = CSharpScript.Run("int c = 1 + 2;");
    var value = state.Variables["c"].Value;

The exception I receive:
    "Could not load type 'Microsoft.CodeAnalysis.MetadataFileReferenceResolver' from assembly 'Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.":"Microsoft.CodeAnalysis.MetadataFileReferenceResolver"}

And the stack trace:
    at Microsoft.CodeAnalysis.Scripting.ScriptOptions..ctor()
    at KatanaMUD.Game.<Run>d__17.MoveNext() in C:\Users\RonPenton\Source\Repos\KatanaMUD\src\KatanaMUD\Game.cs:line 32

I have opened up the nuget package and explored the DLL's using Telerik's JustDecompile. The MetadataFileReferenceResolver class is indeed present in the DLL, but is internal. I didn't think that would cause this error though, so here I am. 

Any thoughts?

**Comment by Pilchie** [comment]
I suspect that this is because the ASP.NET 5 runtime already has it's on version of Microsoft.CodeAnalysis.dll loaded before scripting starts up.

**Comment by RonPenton** [comment]
That appears to be it. The MVC Beta 4 package eventually references CodeAnalysis RC2, which doesn't have the file I need.

Since the DLL's all have the same version, 1.0.0.0, is there anything I can do to reference both DLL's at the same time? Or are we back to DLL Hell with ASP.NET 5?

**Comment by Pilchie** [comment]
Tag @davidfowl.

**Comment by RonPenton** [comment]
Not sure if this makes a difference, but I have been playing around with this a bit more. I got the latest dev assemblies for everything and tried running that, only this time I get a Missing Method Exception for DesktopAssemblyIdentityComparer.Default.

So I looked into this a bit more, and found that the assembly this is being loaded from is:

C:\Users\Ron\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta5-11781\bin\CodeAnalysis.dll

So it looks like the runtime itself is carrying around an older version of CodeAnalysis and ignoring the nuget-included versions.
